src/collections/collection.ts

```diff
@@ -251,51 +205,31 @@ export class Collection {
 
     async findOne(filter: Record<string, any>, options?: FindOneOptions): Promise<Record<string, any> | null> {
         return executeOperation(async (): Promise<Record<string, any> | null> => {
-            const command: FindOneCommand = {
+            const command = {
                 findOne: {
                     filter,
-                    options
+                    options,
+                    ...(options?.sort != null ? { sort: options?.sort } : {}),
+                    ...(options?.projection != null ? { sort: options?.projection } : {}),
                 }
             };
 
-            if (options?.sort) {
-                command.findOne.sort = options.sort;
-                delete options.sort;
```

Comment:
are we missing this (i.e. the `delete` unnecessary fields from options) in the new way of creating the command?